demos/Demo ICR.cpp

```diff
@@ -1,9 +1,504 @@
+    std::cout << "nez: " << neg << std::endl;
+    std::cout <<"average: " << (float)poz/(poz+neg)*100 << "%" << std::endl;
+    
+    
+//    for(int i=0; i<10; i++) {
+//        cout<< "wrong ["<<i<<"] : "<< numrat[i]<<endl;
+//    }
+
+
+
+
+
+    
+    
+    
+
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    
+//      Create the transposed Weight Matrix [36 x 784]
+//        double weightMatrix [hiddenLayer][inputLayer];
+//        for(int i = 0; i < hiddenLayer; i++) {
+//            for(int j = 0; j < inputLayer;  j++) {
+//                weightMatrix[i][j] = myNeuron[j].getWeight(i);
+//                cout<<weightMatrix[i][j]<< " ";
+//            }
+//            cout<<"\n";
+//        }
+
+    
+    
+    
+    //testdatabin[5] = 9 -> 4
+    //[19] = 7 -> 4
+    //[47] = 7 -> 2
+    //[64] = 8 -> 3
+    //[278] = 5-> 6
+    //[441] = 9 -> 7
+    //[784] 6-> 5
+    
+//    int arr2[28][28];
+//    int l=0;
+//
+//        for(int i=0; i<28; i++){
+//            for(int j=0; j<28; j++){
+//                arr2[i][j]=testDataBin[784][l];
+//                l++;
+//            }
+//        }
+//
+//    for(int i = 0; i < 28; i++){
+//        for(int j = 0; j < 28; j++){
+//            arr2[i][j] > 0 ? cout<<"X " : cout<<". ";
+//        }
+//        cout<<endl;
+//    }
+    
+    
+ 
+    
+    
+    
+    
+    
+    
+    
+
+
+//    //std::string image_path = samples::findFile("/Users/diond/Desktop/Pics/a.jpg");
+//    std::string image_path = samples::findFile("/Users/diond/Desktop/foto1.png");
+//    Mat img = imread(image_path, IMREAD_COLOR);
+    
+    
+//
+//for(int z=0; z<1000; z++){
+//
+//    int arr2[28][28];
+//    int l=0;
+//
+//        for(int i=0; i<28; i++){
+//            for(int j=0; j<28; j++){
+//                arr2[i][j]=testDataBin[z][l];
+//                l++;
+//            }
+//        }
+//
+//    if(img.empty())
+//    {
+//        std::cout << "Could not read the image: " << image_path << std::endl;
+//        return 1;
+//    }
+//
+//    cv::Vec3b color = img.at<Vec3b>(Point(0,0));
+//    //std::cout<<color; --> result [255, 249, 250]
+//
+//    for(int i = 0; i < 28; i++)
+//    {
+//        for(int j = 0; j < 28; j++)
+//        {
+//            Vec3b bgrPixel = img.at<Vec3b>(j, i);
+//            //img.at<Vec3b>(i+100, j+20) = 255;
+//            if(arr2[i][j] == 0){
+//                color[0]=255;
+//                color[1]=255;
+//                color[2] =255;
+//            }
+//            else {
+//                color[0]= 255 - arr2[i][j];
+//                color[1]= 255 - arr2[i][j];
+//                color[2]= 255 - arr2[i][j];
+//            }
+//
+//            // set a pixel back to the image
+//
+//            float c =  z/45;
+//            int baba = ceil(c);
+//            img.at<Vec3b>(Point(j+(z*28),i+(c*28))) = color;
+//        }
+//    }
+//
+//    imwrite("/Users/diond/Desktop/foto1.png", img);
+//}
+
+    
+    
+
```

Comment:
This file is too long. The purpose of the demo project is to show how to use the dnn module. Thus it should be as short as possible. Now it has almost half a thousand lines - all the implementation is currently in Demo.cpp file.

Reply:
i also fixed this, and reduced the main file to about 150 lines of code